### PR TITLE
Add new qr code information screen #2968

### PR DESCRIPTION
**Pull request by nickguendling**
## Description
Adds a new information screen before the QR scanner is presented.

Ticket: https://jira.itc.sap.com/browse/EXPOSUREAPP-2968

Simulator Screen Shot - iPhone 11 Pro - 2020-10-07 at 16 45 51

### CODE REVIEW

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import UIKit
+
+extension String {
+
+	func inserting(string: String, style: ENAFont) -> NSAttributedString {
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:22]: 🤔`inserting` would suggest that the string only gets something added, but some formatting seems to happen here as well?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift [hunk 2]
```diff
@@ -102,11 +102,6 @@ extension ExposureSubmissionCoordinator {
 		setupDismissConfirmationOnSwipeDown(for: vc)
 	}
 
-	private func present(_ vc: UIViewController) {
-		self.navigationController?.present(vc, animated: true)
-		setupDismissConfirmationOnSwipeDown(for: vc)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:107]: Just checking; Dismissconfirmation still working?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/QRScanner/ExposureSubmissionQRScannerFocusView.swift [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+@IBDesignable
+final class ExposureSubmissionQRScannerFocusView: UIView {
+	@IBInspectable var backdropOpacity: CGFloat = 0
+	@IBInspectable var cornerRadius: CGFloat = 0
+	@IBInspectable var borderWidth: CGFloat = 1
+
+	override func prepareForInterfaceBuilder() {
+		super.prepareForInterfaceBuilder()
+
+		backgroundColor = UIColor(white: 1, alpha: 0.5)
```
Comment (marcussc) [inline src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/QRScanner/ExposureSubmissionQRScannerFocusView.swift:29]: Darkmode safe?

**Review by marcussc — COMMENTED** [review]
Nice drive by refactor 💫

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift (hunk 2 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:107]: Yep, still works 👍

Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/ExposureSubmissionCoordinator.swift:107]: +1

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift (hunk 1 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:22]: Do you have a name suggestion?

File: src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/QRScanner/ExposureSubmissionQRScannerFocusView.swift (hunk 3 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Scenes/ExposureSubmission/View/Controller/QRScanner/ExposureSubmissionQRScannerFocusView.swift:29]: Yep, this screen is always on the dark side 🕶️

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift (hunk 1 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:22]:
```suggestion
	func inserting(_ string: String, as: ENAFont) -> NSAttributedString {
```
Then the call side would look like this:
`AppStrings.ExposureSubmissionDispatch.hotlineButtonDescription.inserting(AppStrings.ExposureSubmissionDispatch.positiveWord, as: .headline)`

Comment (KaiTeuber) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:22]: Is the swift keyword 'as' usable?

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import UIKit
+
+extension String {
+
```
Comment (whiskey) [inline src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:21]: Documentation please.

**Review by KaiTeuber — APPROVED** [review]

**Review by janetback — APPROVED** [review]
UA approved

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift (hunk 1 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:22]: Damn you are right, hmm.
But do you agree that this function deserves a better naming our not? :)

Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift]: We narrowed down the functionality now and renamed it to `inserting(emphasizedString:)`. We think that makes it much clearer. Do you agree?

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift (hunk 4 above)
Comment (nickguendling) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift:21]: Added 😁👍

File: src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift (hunk 1 above)
Comment (marcussc) [reply src/xcode/ENA/ENA/Source/Extensions/String+Insertion.swift]: Lovely 💫

**Review by marcussc — APPROVED** [review]

**Pull request merged**